### PR TITLE
Wishlist feature

### PR DESCRIPTION
**Pull request by rushikesh3299**
Wishlist feature 
Add to wishlist
remove from wishlist
wishlist to cart

### CODE REVIEW

File: src/context/wishlist-context.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+  };
+
+  const checkItemInWiishlist = (itemId) => {
+    let itemPresent = false;
+    wishlistItems.forEach((item) => {
+      if (item._id == itemId) {
+        itemPresent = true;
+      }
+    });
```
Comment (JantuDeb) [inline src/context/wishlist-context.js]: `const itemPresnet = wishlistItems.some(item=>item._id===itemId)`

**Review by JantuDeb — APPROVED** [review]
Overall this is good

File: src/App.js [hunk 2]
```diff
@@ -26,6 +27,7 @@ export const App = () => {
         <Route path="/login" element={<Login />} />
         <Route path="/signup" element={<Signup />} />
         <Route path="/cart" element={<Cart />}></Route>
+        <Route path="/wishlist" element={<Wishlist />}></Route>
```
Comment (MiheerTamkhane) [inline src/App.js]: Use self closing tags here

File: src/components/products/Products.js [hunk 3]
```diff
@@ -1,14 +1,19 @@
 import "./products.css";
 import { Rating } from "../index";
-import { useProduct, useLogin, useCart } from "../../context/index";
+import {
+  useProduct,
+  useLogin,
+  useCart,
+  useWishlist,
+} from "../../context/index";
```
Comment (MiheerTamkhane) [inline src/components/products/Products.js:8]: no need to add index at last it takes by default

File: src/pages/wishlistpage/Wishlist.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+              </div>
+              <div className="product-card-discription">
+                <div className="product-card-name">{item.title}</div>
+                <div className="product-card-brand">{item.brand}</div>
+                <div className="product-card-price">
+                  <div>{item.price}</div>
+                  <div>(10% OFF)</div>
+                </div>
+                <div className="product-card-quantity">{item.weight}</div>
+                <button className="product-card-btn-add">Add to cart</button>
```
Comment (MiheerTamkhane) [inline src/pages/wishlistpage/Wishlist.js:33]: better you destructure item here.

**Review by MiheerTamkhane — APPROVED** [review]
Code is clean and readable and also working fine
good to merge

File: src/components/products/Products.js (hunk 3 above)
Comment (rushikesh3299) [reply src/components/products/Products.js:8]: Thanks. I am unaware about this

**Pull request merged**